### PR TITLE
Drop the update.title field from the database

### PR DESCRIPTION
**Pull request by bowlofeggs**
There are two commits in this pull request. Please see their messages for details. The summary:

* The first one indexes a field that really should have always been indexed. This is needed for the downgrade() on the migration for the second commit, which takes more than 10 minutes without an index (I killed it, so no idea how long it actually takes, just that it's more than 10 minutes). With the index, the downgrade() takes about 7 seconds on my laptop.
* The second commit drops the title field from updates, and disallows referencing updates by title.

### CODE REVIEW

**Comment by bowlofeggs** [comment]
This pull request will not pass CI until #2987 is merged, as it uses some Python 3-only syntax in the CLI.

**Review by Zlopez — APPROVED** [review]
LGTM

File: bodhi/server/models.py [hunk 1]
```diff
@@ -1672,7 +1666,7 @@ class Update(Base):
     date_stable = Column(DateTime)
 
     # eg: FEDORA-EPEL-2009-12345
-    alias = Column(Unicode(32), unique=True, nullable=True)
+    alias = Column(Unicode(32), unique=True, nullable=False)
```
Comment (sebwoj) [inline bodhi/server/models.py:1664]: With this change, we can mark commit as fixing #1714

File: bodhi/tests/server/test_models.py [hunk 2]
```diff
@@ -3248,7 +3246,7 @@ def test_comment_empty(self):
         self.assertEqual(str(exc.exception), 'You must provide either some text or feedback')
 
     def test_get_url(self):
-        self.assertEqual(self.obj.get_url(), u'updates/TurboGears-1.0.8-3.fc11')
+        self.assertEqual(self.obj.get_url(), f'updates/{self.obj.alias}')
         idx = 'a3bbe1a8f2'
         with mock.patch(target='uuid.uuid4', return_value='wat'):
             self.obj.assign_alias()
```
Comment (sebwoj) [inline bodhi/tests/server/test_models.py]: Method `assign_alias` was removed.

File: bodhi/tests/server/test_models.py [hunk 3]
```diff
@@ -3561,14 +3559,14 @@ def test_validate_release_failure(self):
...
         update = model.Update(
-            title=u'testmodule-master-2',
             builds=[build], user=user,
             status=UpdateStatus.testing,
             request=UpdateRequest.stable,
             type=UpdateType.enhancement,
             notes=u'Useful details!',
             test_gating_status=TestGatingStatus.passed)
         update.release = release
+        update.assign_alias()
```
Comment (sebwoj) [inline bodhi/tests/server/test_models.py]: Method `assign_alias` was removed.

File: bodhi/tests/server/test_models.py [hunk 4]
```diff
@@ -3614,22 +3612,22 @@ def test_validate_release_success(self):
                                    package=package)
         self.db.add(build2)
         update1 = model.Update(
-            title=u'testmodule-master-2',
             builds=[build1], user=user,
             status=UpdateStatus.testing,
             request=UpdateRequest.stable,
             notes=u'Useful details!', release=release,
             test_gating_status=TestGatingStatus.passed)
+        update1.assign_alias()
```
Comment (sebwoj) [inline bodhi/tests/server/test_models.py]: Method `assign_alias` was removed.

File: bodhi/tests/server/test_models.py [hunk 5]
```diff
@@ -3614,22 +3612,22 @@ def test_validate_release_success(self):
...
 
         # This should not raise an Exception.
         update2 = model.Update(
-            title=u'testmodule-master-2',
             builds=[build2], user=user,
             status=UpdateStatus.testing,
             request=UpdateRequest.stable,
             notes=u'Useful details!', release=release,
             test_gating_status=TestGatingStatus.passed)
+        update2.assign_alias()
```
Comment (sebwoj) [inline bodhi/tests/server/test_models.py]: Method `assign_alias` was removed.

File: bodhi/server/models.py [hunk 6]
```diff
@@ -1631,10 +1627,8 @@ class Update(Base):
 
     __tablename__ = 'updates'
     __exclude_columns__ = ('id', 'user_id', 'release_id')
-    __include_extras__ = ('meets_testing_requirements', 'url',)
-    __get_by__ = ('title', 'alias')
```
Comment (sebwoj) [inline bodhi/server/models.py:1630]: Removing `title` from this container will block access to updates via `/updates/{title}`(like: bodhi.fedoraproject.org/updates/phpdoc-2.9.0-7.fc29)  so maybe we should add note in release notes in B/I section?

**Review by sebwoj — CHANGES_REQUESTED** [review]
It looks like this PR is in WIP state, but I've left just a few comments.

File: bodhi/server/models.py (hunk 1 above)
Comment (bowlofeggs) [reply bodhi/server/models.py:1664]: Fixed, and thanks!

File: bodhi/tests/server/test_models.py (hunk 2 above)
Comment (bowlofeggs) [reply bodhi/tests/server/test_models.py]: Yeah, I believe I've successfully removed this everywhere.

File: bodhi/server/models.py (hunk 6 above)
Comment (bowlofeggs) [reply bodhi/server/models.py:1630]: Added!

**Comment by bowlofeggs** [comment]
The ```f28-integration``` test failure is a timeout waiting on the DB container to start. I wrote #3054 to help with this, and I think we can safely ignore it here.

**Comment by bowlofeggs** [comment]
This patch is planned for inclusion in the upcoming 4.0.0 release: #3221

**Comment by bowlofeggs** [comment]
Bodhi 4.0.0 beta is built and deployed to staging:

https://copr.fedorainfracloud.org/coprs/bowlofeggs/bodhi-pre-release
https://bodhi.stg.fedoraproject.org/

**Pull request merged**